Since PR 2574 switch to use HexFormat.of() in JCEEncryptor, plain text password from properties is no longer working

Since PR #2574, attempting to connect using plaintext password is not working. The reason is that the switch from homegrown hex decoding to `HexFormat.of()` makes a different exception (`IllegalStateException`) when the decoding fails, and it does not go to the correct exception block.

The code in question is here, from `JCEEncryptor`:
```java
public synchronized String decryptPassword(String encryptedPswd) {
        if (encryptedPswd == null) {
            return null;
        }

        String password = null;
        byte[] input = null;
        byte[] bytePassword = null;

        try {
            input = HexFormat.of().parseHex(encryptedPswd);
            SecretKey skGCM = Synergizer.getAESGCMMultitasker();
            byte[] ivGCM = new byte[IV_GCM_LENGTH];
            System.arraycopy(input, 0, ivGCM, 0, IV_GCM_LENGTH);
            AlgorithmParameterSpec parameterSpecGCM = new GCMParameterSpec(128, ivGCM);
            bytePassword = new byte[input.length - IV_GCM_LENGTH];
            System.arraycopy(input, IV_GCM_LENGTH, bytePassword, 0, input.length - IV_GCM_LENGTH);
            decryptCipherAES_GCM.init(Cipher.DECRYPT_MODE, skGCM, parameterSpecGCM);
            // try AES/GCM first
            password = new String(decryptCipherAES_GCM.doFinal(bytePassword), StandardCharsets.UTF_8);
        } catch (ArrayIndexOutOfBoundsException | IllegalBlockSizeException | NumberFormatException ce) {
            // buildBytesFromHexString failed, assume clear text
            password = encryptedPswd;
        } catch (Exception u) {
                throw ValidationException.errorDecryptingPasswordOldAlgorithm(u);
        }
        return password;
    }
```

In this particular case, there is fallback behavior intended, but since `IllegalStateException` is thrown, it falls to the validation error instead of the clear text path.

Here is the stack trace when this happens:
```
Exception [EclipseLink-7360] (Eclipse Persistence Services - 5.0.0-B13.v202512111447-a92ecc697995fa55709edc3c28d855c9c3ac2a1d): org.eclipse.persistence.exceptions.ValidationException
Exception Description: Database password was encrypted by deprecated algorithm. Reencrypt it by `passwordUpdate.sh` from eclipselink.zip bundle.
Internal Exception: java.lang.IllegalArgumentException: string length not even: 9
jakarta.persistence.PersistenceException: Exception [EclipseLink-7360] (Eclipse Persistence Services - 5.0.0-B13.v202512111447-a92ecc697995fa55709edc3c28d855c9c3ac2a1d): org.eclipse.persistence.exceptions.ValidationException
Exception Description: Database password was encrypted by deprecated algorithm. Reencrypt it by `passwordUpdate.sh` from eclipselink.zip bundle.
Internal Exception: java.lang.IllegalArgumentException: string length not even: 9
	at org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.deploy(EntityManagerSetupImpl.java:872)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.getAbstractSession(EntityManagerFactoryDelegate.java:233)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:340)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:495)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:458)
...
Caused by: Exception [EclipseLink-7360] (Eclipse Persistence Services - 5.0.0-B13.v202512111447-a92ecc697995fa55709edc3c28d855c9c3ac2a1d): org.eclipse.persistence.exceptions.ValidationException
Exception Description: Database password was encrypted by deprecated algorithm. Reencrypt it by `passwordUpdate.sh` from eclipselink.zip bundle.
Internal Exception: java.lang.IllegalArgumentException: string length not even: 9
	at org.eclipse.persistence.exceptions.ValidationException.errorDecryptingPasswordOldAlgorithm(ValidationException.java:919)
	at org.eclipse.persistence.internal.security.JCEEncryptor.decryptPassword(JCEEncryptor.java:112)
	at org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.updateLogins(EntityManagerSetupImpl.java:2544)
	at org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.updateSession(EntityManagerSetupImpl.java:2867)
	at org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.deploy(EntityManagerSetupImpl.java:763)
	... 40 more
Caused by: java.lang.IllegalArgumentException: string length not even: 9
	at java.base/java.util.HexFormat.parseNoDelimiter(HexFormat.java:845)
	at java.base/java.util.HexFormat.parseHex(HexFormat.java:534)
	at java.base/java.util.HexFormat.parseHex(HexFormat.java:502)
	at org.eclipse.persistence.internal.security.JCEEncryptor.decryptPassword(JCEEncryptor.java:98)
	... 43 more
```

Reproduce with a connection set up via properties, like the following:
```
jakarta.persistence.jdbc.driver=oracle.jdbc.OracleDriver
jakarta.persistence.jdbc.url=jdbc:oracle:thin:@1.2.3.4:1521:ORCL
jakarta.persistence.jdbc.user=sqltester
jakarta.persistence.jdbc.password=sqltester
```

- EclipseLink version: 5.0.0-B13
- Java/JDK version: 25.0.2

This can be worked around by providing an equivalent of the pass through encryptor, e.g.
```java
public class OracleTestingPassThroughEncryptor implements Securable {
    @Override
    public String encryptPassword(String pswd) {
        return pswd;
    }

    @Override
    public String decryptPassword(String encryptedPswd) {
        return encryptedPswd;
    }
}
```

and using it as such:
```
eclipselink.login.encryptor=com.foo.OracleTestingPassThroughEncryptor
```

I believe this is a bug because there's clearly intended fallback behavior and it's not triggering because of the alternate exception.

--- a/foundation/eclipselink.core.test/src/test/java/org/eclipse/persistence/testing/tests/junit/security/SecurableBackwardsCompatibilityTest.java
+++ b/foundation/eclipselink.core.test/src/test/java/org/eclipse/persistence/testing/tests/junit/security/SecurableBackwardsCompatibilityTest.java
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2026 Contributors to the Eclipse Foundation. All rights reserved.
  * Copyright (c) 1998, 2025 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -126,6 +127,9 @@ public class SecurableBackwardsCompatibilityTest {
         Securable securable = new JCEEncryptor();
         String decryptedString = securable.decryptPassword(plainTextString);
         Assert.assertEquals("Passwords should match.", plainTextString, decryptedString);
+
+        decryptedString = securable.decryptPassword("sqltester");
+        Assert.assertEquals("Passwords should match.", "sqltester", decryptedString);
     }
 
     /**

--- a/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/security/JCEEncryptor.java
+++ b/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/security/JCEEncryptor.java
@@ -1,6 +1,6 @@
 /*
+ * Copyright (c) 2024, 2026 Contributors to the Eclipse Foundation. All rights reserved.
  * Copyright (c) 1998, 2025 Oracle and/or its affiliates. All rights reserved.
- * Copyright (c) 2024 Contributors to the Eclipse Foundation. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Public License v. 2.0 which is available at
@@ -105,7 +105,7 @@ public final class JCEEncryptor implements org.eclipse.persistence.security.Secu
             decryptCipherAES_GCM.init(Cipher.DECRYPT_MODE, skGCM, parameterSpecGCM);
             // try AES/GCM first
             password = new String(decryptCipherAES_GCM.doFinal(bytePassword), StandardCharsets.UTF_8);
-        } catch (ArrayIndexOutOfBoundsException | IllegalBlockSizeException | NumberFormatException ce) {
+        } catch (ArrayIndexOutOfBoundsException | IllegalBlockSizeException | IllegalArgumentException ce) {
             // buildBytesFromHexString failed, assume clear text
             password = encryptedPswd;
         } catch (Exception u) {

